collect_app/src/main/java/org/odk/collect/android/utilities/DownloadFormListUtils.java

```diff
@@ -77,14 +76,14 @@ public DownloadFormListUtils() {
         SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(
                 Collect.getInstance().getBaseContext());
 
-        // Remove trailing '/'
-        while (url != null && url.endsWith("/")) {
-            url = url.substring(0, url.length() - 1);
-        }
-
         String downloadListUrl = url != null ? url :
                 settings.getString(GeneralKeys.KEY_SERVER_URL,
                         Collect.getInstance().getString(R.string.default_server_url));
+
+        if (downloadListUrl.endsWith("/")) {
```

Comment:
The rationale for the while loop was presumably that since it's user input there could be multiple training /, as unlikely as that might seem. Did you change it because it just doesn't seem likely or is there another reason?

Reply:
Nope that is a genuine mistake. I can look at putting that back in and make sure it's covered by  a test so it doesn't get absentmindedly removed again by someone (like me).